ULogViewer/Logs/DataSources/StandardOutputLogDataSource.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.Logging;
+
+namespace CarinaStudio.ULogViewer.Logs.DataSources
+{
+	/// <summary>
+	/// Implementation of <see cref="StandardOutputLogDataSource"/>.
+	/// </summary>
+	class StandardOutputLogDataSource : BaseLogDataSource
+	{
+		// Fields.
+		string? commandFileOnReady;
+		readonly LogDataSourceOptions options;
+
+
+		/// <summary>
+		/// Initialize new <see cref="StandardOutputLogDataSource"/> instance.
+		/// </summary>
+		/// <param name="provider"><see cref="StandardOutputLogDataSourceProvider"/> which creates this instane.</param>
+		/// <param name="options"><see cref="LogDataSourceOptions"/> to create source.</param>
+		internal StandardOutputLogDataSource(StandardOutputLogDataSourceProvider provider, LogDataSourceOptions options) : base(provider, options)
+		{
+			this.options = options;
+		}
+
+
+		// Open reader core.
+		protected override LogDataSourceState OpenReaderCore(out TextReader? reader)
+		{
+			if (commandFileOnReady == null)
+			{
+				reader = null;
+				return LogDataSourceState.SourceNotFound;
+			}
+			using var process = new Process();
+			process.StartInfo.FileName = commandFileOnReady;
+			process.StartInfo.UseShellExecute = false;
+			process.StartInfo.RedirectStandardOutput = true;
+			process.Start();
+			reader = process.StandardOutput;
+			return LogDataSourceState.ReaderOpened;
+		}
+
+
+		// Prepare core.
+		protected override LogDataSourceState PrepareCore()
+		{
+			if (string.IsNullOrEmpty(options.Command))
+				return LogDataSourceState.SourceNotFound;
+			var regex = new Regex("^(?<ExecutableCommand>([^\\s\"]*)|\"([^\\s])*\")[\\s$]");
```

Comment:
It is better to be ```static readonly``` field because it is same for all cases and building ```Regex``` costs long time.